### PR TITLE
Remove error package and cause functions

### PR DESCRIPTION
**Pull request by kannappanr**
## Description
Delete error package. Remove `cause` function used in the code.

## Motivation and Context
Part of refactoring of server side logging.

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/object_api_suite_test.go [hunk 1]
```diff
@@ -756,7 +755,6 @@ func testGetDirectoryReturnsObjectNotFound(obj ObjectLayer, instanceType string,
 	for i, testCase := range testCases {
 		_, expectedErr := obj.GetObjectInfo(context.Background(), bucketName, testCase.dir)
 		if expectedErr != nil {
```
Comment (ebozduman) [inline cmd/object_api_suite_test.go]: After removing the `errors.Cause(expectedErr)`, we can combine the 2 if clauses as:
`if expectedErr != nil && expectedErr.Error() != testCase.err.Error() {`

File: cmd/xl-v1-healing.go [hunk 2]
```diff
@@ -116,13 +115,13 @@ func healBucket(ctx context.Context, storageDisks []StorageAPI, bucket string, w
...
 			if _, serr := disk.StatVol(bucket); serr != nil {
-				if errors.Cause(serr) == errDiskNotFound {
+				if serr == errDiskNotFound {
 					beforeState[index] = madmin.DriveStateOffline
 					afterState[index] = madmin.DriveStateOffline
 					dErrs[index] = serr
 					return
 				}
-				if errors.Cause(serr) != errVolumeNotFound {
+				if serr != errVolumeNotFound {
```
Comment (ebozduman) [inline cmd/xl-v1-healing.go:129]: could have been 
`if` => `else if`

**Review by ebozduman — CHANGES_REQUESTED** [review]

File: cmd/xl-v1-healing.go (hunk 2 above)
Comment (kannappanr) [reply cmd/xl-v1-healing.go:129]: I think the existing code is cleaner to read.

**Review by ebozduman — DISMISSED** [review]

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#bab42c14-37ec-49f7-813f-1f260b57d101) |
<a name="bab42c14-37ec-49f7-813f-1f260b57d101"></a>
### 5784-34db13e/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:31472
ACCESS_KEY:      minioazure1
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp e09e27261f80:/mint/log /tmp/mint-logs'
(1/12) Running aws-sdk-go tests ... done in 1 seconds
(2/12) Running aws-sdk-php tests ... done in 7 minutes and 56 seconds
(3/12) Running aws-sdk-ruby tests ... done in 39 seconds
(4/12) Running awscli tests ... done in 4 minutes and 7 seconds
(5/12) Running mc tests ... done in 7 minutes and 26 seconds
(6/12) Running minio-dotnet tests ... done in 1 minutes and 17 seconds
(7/12) Running minio-go tests ... done in 1 seconds
(8/12) Running minio-java tests ... done in 37 seconds
(9/12) Running minio-js tests ... FAILED in 2 minutes and 34 seconds
{
  "name": "minio-js",
  "function": "getBucketPolicy(bucketName, cb)",
  "args": "bucketName:minio-js-test-199425d5-4db5-4254-982c-8e45a39468ce",
  "duration": 261,
  "status": "FAIL",
  "error": "S3Error: The bucket policy does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:56:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:117:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:48:26) at DestroyableTransform.<anonymous> (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:135:12) at finishMaybe (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:371:12) at endWritable (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:378:3) at DestroyableTransform.Writable.end (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:356:5) at IncomingMessage.onend (_stream_readable.js:511:10) at endReadableNT (_stream_readable.js:974:12) at _combinedTickCallback (internal/process/next_tick.js:80:11) at process._tickCallback (internal/process/next_tick.js:104:9)"
}

Executed 8 out of 12 tests successfully.
```

**Comment by balamurugana** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#fd28408e-d3d5-4825-a6f1-ff66e0c87954) |
<a name="fd28408e-d3d5-4825-a6f1-ff66e0c87954"></a>
### 5784-6d78394/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT: minikube:30919
ACCESS_KEY:      minioazure1
SECRET_KEY:      ***REDACTED***
ENABLE_HTTPS:    0
SERVER_REGION:   us-east-1
MINT_DATA_DIR:   /mint/data
MINT_MODE:       core

To get logs, run 'docker cp 0b107083083a:/mint/log /tmp/mint-logs'
(1/12) Running aws-sdk-go tests ... done in 1 seconds
(2/12) Running aws-sdk-php tests ... done in 8 minutes and 2 seconds
(3/12) Running aws-sdk-ruby tests ... done in 58 seconds
(4/12) Running awscli tests ... done in 4 minutes and 0 seconds
(5/12) Running mc tests ... done in 7 minutes and 23 seconds
(6/12) Running minio-dotnet tests ... done in 1 minutes and 9 seconds
(7/12) Running minio-go tests ... done in 1 seconds
(8/12) Running minio-java tests ... done in 33 seconds
(9/12) Running minio-js tests ... FAILED in 2 minutes and 50 seconds
{
  "name": "minio-js",
  "function": "getBucketPolicy(bucketName, cb)",
  "args": "bucketName:minio-js-test-323ca63e-7bc1-49e3-93f0-6ea5dca73797",
  "duration": 253,
  "status": "FAIL",
  "error": "S3Error: The bucket policy does not exist at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:56:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:117:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:48:26) at DestroyableTransform.<anonymous> (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:135:12) at finishMaybe (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:371:12) at endWritable (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:378:3) at DestroyableTransform.Writable.end (node_modules/minio/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:356:5) at IncomingMessage.onend (_stream_readable.js:511:10) at endReadableNT (_stream_readable.js:974:12) at _combinedTickCallback (internal/process/next_tick.js:80:11) at process._tickCallback (internal/process/next_tick.js:104:9)"
}

Executed 8 out of 12 tests successfully.
```

**Review by harshavardhana — APPROVED** [review]

**Review by ebozduman — APPROVED** [review]

**Pull request merged**